### ISSUE TITLE
Deprecation

### ISSUE THREAD

**Issue by m59peacemaker** [ct0r/fetchware#1]
What led you to the decision to use `reduce` instead of a middleware pattern? It doesn't limit decorators in any way?

**Comment by ct0r** [comment]
Hi, this library was supposed to be support utility for other fetchware packages and I have decided to compose them via reduce, because you don't need to have dependency to this package. It's also more compatible with js decorators.

**Comment by m59peacemaker** [comment]
Right. I understand. I like the idea of reduce for sure. The difference between this approach is that it doesn't cascade, which is that the request is processed downward through the middleware in the given order, and the response is processed in reverse, upward through the given order. I'm not sure why other middleware things use cascading like that and if it would offer anything for fetch. I was hoping you might know.
